examples/ola-recorder.cpp

```diff
@@ -115,36 +121,51 @@ int VerifyShow(const string &filename) {
     return ola::EXIT_NOINPUT;
 
   map<unsigned int, unsigned int> frames_by_universe;
-  uint64_t total_time = 0;
 
-  unsigned int universe;
-  ola::DmxBuffer buffer;
-  unsigned int timeout;
+  ShowEntry entry;
   ShowLoader::State state;
+  unsigned int playback_pos = 0;
+  bool playing = false;
   while (true) {
-    state = loader.NextFrame(&universe, &buffer);
+    state = loader.NextEntry(&entry);
     if (state != ShowLoader::OK)
       break;
-    frames_by_universe[universe]++;
-
-    state = loader.NextTimeout(&timeout);
-    if (state != ShowLoader::OK)
+    playback_pos += entry.next_wait;
+    if (FLAGS_stop > 0 && playback_pos >= FLAGS_stop) {
+      // Compensate for overshooting the stop time
+      playback_pos -= (playback_pos - FLAGS_stop);
       break;
-    total_time += timeout;
+    }
+    if (!playing && playback_pos >= FLAGS_start) {
+      // Found the start point
+      playing = true;
+      entry.next_wait = playback_pos - FLAGS_start;
+    }
+    if (playing) {
+      frames_by_universe[entry.universe]++;
+    }
   }
 
+  const unsigned int total_time = playback_pos - FLAGS_start;
   map<unsigned int, unsigned int>::const_iterator iter;
   unsigned int total = 0;
   cout << "------------ Summary ----------" << endl;
+  if (FLAGS_start > 0) {
+    cout << "Starting at: " << FLAGS_start / 1000.0 << " second(s)" << endl;
+    if (FLAGS_stop > 0) {
+      cout << "Stopping at: " << FLAGS_stop / 1000.0 << " second(s)" << endl;
+    }
+    std::cout << endl;
+  }
   for (iter = frames_by_universe.begin(); iter != frames_by_universe.end();
        ++iter) {
     cout << "Universe " << iter->first << ": " << iter->second << " frames" <<
       endl;
     total += iter->second;
   }
+  std::cout << endl;
   cout << "Total frames: " << total << endl;
-  cout << "Playback time: " << total_time / 1000 << "." << total_time % 10 <<
-    " seconds" << endl;
+  cout << "Playback time: " << total_time / 1000.0 << " second(s)" << endl;
```

Comment:
Yes, I'd noticed this when testing duration and getting confused. It might be worth us looking at setw to make something nicer output wise.

Reply:
I thought there was a bug when `--start 1000 --end 1500` resulted in some weird output... :grin: